src/qml/FeatureForm.qml

```diff
@@ -300,11 +301,15 @@ Page {
         Connections {
           target: form
           onAboutToSave: {
-            try {
-              attributeEditorLoader.item.pushChanges()
+            // it may not be implemented
+            if ( attributeEditorLoader.item.pushChanges ) {
+              attributeEditorLoader.item.pushChanges( form.model.featureModel.feature )
             }
-            catch ( err )
-            {}
+          }
+          onValueChanged: (field, oldValue, newValue) => {
+            // it may not be implemented
+            if ( attributeEditorLoader.item.siblingValueChanged )
+              attributeEditorLoader.item.siblingValueChanged( field, form.model.featureModel.feature )
```

Comment:
One question: why are you passing on a field parameter here which doesn't seem to be used afterwards?

Reply:
So, thinking about it some more, I'd remove the field parameter (which isn't used), and rename the function to something like featureFormChanged. I think it'd help readability.

Reply:
For this particular use case the "field" parameter is ignored indeed. However, the value relation widget will update only if the the field that updated is the one used in the expression `current_value( 'attr_name' )` if I was more strict. The only way to see who triggered the callback is to pass the field parameter. Nevertheless, I am not using it in this particular case, as the performance gain does not outweight the code complexity involved.

In the end I decided to leave the param, as I though it would be nice to have for any future case like this in the widget implementations. No strong opinions though.

Reply:
No strong opinion on my side either, and you seem to have thought this through. I'm satisfied :wink: